### PR TITLE
Feature/historic results component

### LINKED ISSUE #198: Historic results component does not query Forms endpoint
Currently, the component displaying history of forms completed does not query the Forms endpoint and shows only dummy data.

### PR DESCRIPTION
**Pull request by irinel-nistor**
### What does it fix?

Closes #198 

- made request to the forms endpoint
- build profile history based on form api response
- populated the profile page with the history

### How has it been tested?

Did manual test for user profile and family members

**Issues**
*The form question asks only if you had temperature over 38, not it's actual value
image
* Values in the symptoms table seem to be a bit off for input: 
[  {    "id": 113, "date": 1588194000, "soreThroat": true, "cough": true, "shortnessBreath": true, "runningNose": false  },
  {  "id": 116, "date": 1588021200, "soreThroat": false, "cough": false, "shortnessBreath": false, "runningNose": true  },
  {   "id": 117, "date": 1588626000, "soreThroat": true, "cough": true,"shortnessBreath": true, "runningNose": true  },
  {    "id": 120,    "date": 1588712400,    "soreThroat": false,    "cough": false,    "shortnessBreath": false,    "runningNose": true  }
]
image
* It's not clear for me if the "Istoric deplasări" section can be populated using the current version of the form data.

### CODE REVIEW

**Comment by irinel-nistor** [comment]
@teodoradima  was a little busy, so we worked together on this one

File: frontend/src/components/Account/common/ProfileHistory/ProfileHistoryBuilder.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    soreThroatDate,
+    hadCough,
+    coughDate,
+    hadShortnessBreath,
+    shortnessBreathDate,
+    hadRunningNose,
+    runningNoseDate,
+    hadOtherSymptoms,
+    otherSymptomsDescription
+  ] = answers;
```
Comment (surdu) [inline frontend/src/components/Account/common/ProfileHistory/ProfileHistoryBuilder.js]:
```suggestion
  const [
    hadFever,
    feverDate,
    hadSoreThroat,
    soreThroatDate,
    hadCough,
    coughDate,
    hadShortnessBreath,
    shortnessBreathDate,
    hadRunningNose,
    runningNoseDate,
    hadOtherSymptoms,
    otherSymptomsDescription
  ] = answers.slice(12);
```

**Review by surdu — CHANGES_REQUESTED** [review]

File: frontend/src/api/evaluationApi.js [hunk 2]
```diff
@@ -12,7 +12,13 @@ const EvaluationApi = {
   sendSelfEvaluationResults: formResults => api.post("/form", formResults),
 
   sendEvaluationResults: (formResults, id) =>
-    api.post("/form", formResults, { params: { id } })
+    api.post("/form", formResults, { params: { id } }),
+
+  getEvaluationResults: async userId => {
+    const { data } = await api.get(`form?id=${userId}`);
```
Comment (tudorgergely) [inline frontend/src/api/evaluationApi.js]:
```suggestion
    const { data } = await api.get(`/form?id=${userId}`);
```
It's a bit more clear this way

File: frontend/src/components/Account/common/ProfileHistory/ProfileHistory.js [hunk 3]
```diff
@@ -20,44 +23,58 @@ import {
...
+  const [evaluationResults, setEvaluationResults] = useState();
+  let history = { temperature: [] };
+
+  useEffect(() => {
+    EvaluationApi.getEvaluationResults(data.id).then(res => {
+      setEvaluationResults(res);
+    });
+  }, []);
+
+  if (evaluationResults) history = buildHistory(evaluationResults);
```
Comment (tudorgergely) [inline frontend/src/components/Account/common/ProfileHistory/ProfileHistory.js]:
```suggestion
  const [history, setHistory] = useState({ temperature: [] });

  useEffect(() => {
    EvaluationApi.getEvaluationResults(data.id).then(res => {
      if (res) {
        setHistory(buildHistory(res));
      }
    });
  }, []);
```

File: frontend/src/components/Account/common/ProfileHistory/ProfileHistoryBuilder.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  }
+
+  const date = new Date(symptomParameter.symptomDate.answer).getTime() / 1000;
+  const symptomOnDate = symptoms.find(s => s.date === date);
+
+  if (symptomOnDate) {
+    symptomOnDate[symptomParameter.symptomName] = true;
+  } else {
+    let symptom = {
+      id: id,
```
Comment (tudorgergely) [inline frontend/src/components/Account/common/ProfileHistory/ProfileHistoryBuilder.js:106]: what do you need the id for?

Comment (irinel-nistor) [reply frontend/src/components/Account/common/ProfileHistory/ProfileHistoryBuilder.js:106]: https://taskforce-fe-components.now.sh/?path=/story/completed-forms--completed-forms. It is probably used to set the component key

File: frontend/src/components/Account/common/ProfileHistory/ProfileHistory.js (hunk 3 above)
Comment (irinel-nistor) [reply frontend/src/components/Account/common/ProfileHistory/ProfileHistory.js]: Nice suggestion, thanks.

**Review by surdu — DISMISSED** [review]

**Comment by RaduCStefanescu** [comment]
I didn't check the code, but what happens in the case the user has not filled yet any assessments? This happens at the first login, after the user filled in his profile.
In this case, we should not display the historic assessments component.

**Comment by irinel-nistor** [comment]
@RaduCStefanescu I added a check, thanks for the suggestion.

**Review by surdu — APPROVED** [review]

**Review by RaduCStefanescu — APPROVED** [review]

**Pull request merged**